woocommerce/woocommerce-ios
# Add Product Variations: List Existing Attributes

Fixes #3181 

## Description
After having implemented the [Networking layer](https://github.com/woocommerce/woocommerce-ios/pull/3290), and after the changes to the [Storage and Yosemite layers](https://github.com/woocommerce/woocommerce-ios/pull/3299) to handle global Product Attributes, in this PR we display the existing local and global Attributes in the new "Add Attribute" screen, part of the Add Product Variation flow (M5).


## Changes
- Now in Storage we delete unused Product Attributes when we synchronize all the global Product Attributes when they are not connected to a product.
- Now it’s possible to load all the `ProductAttribute`s for a specific `siteID`.
- Implemented the `AddAttributeViewController` that display the product attributes plus a text field on top where the user can add a new attribute.
- Implemented `AddAttributeViewModel` that provides view data for the Add Attribute screen, and handles init/UI/navigation actions needed. + unit tests
- Added `ProductAttributeStore` inside the list of `AuthenticatedState`.

## Testing
1. Open a Variable Product without variations.
2. Tap the cell "Add variations".
3. The new screen "Add Attribute" should be presented.
4. In the "Add Attribute" screen you should see the list of global attributes + local product attributes. No other actions are allowed at the moment.

## Screenshot
<img src="https://user-images.githubusercontent.com/495617/101928377-cbe83e00-3bd5-11eb-8ecb-c85dfaf094f9.png" width=300 />




Update release notes:
